core/src/main/java/com/bumble/appyx/core/node/Node.kt

```diff
@@ -95,16 +99,37 @@ abstract class Node(
             LocalNode provides this,
             LocalLifecycleOwner provides this,
         ) {
-            DerivedSetup()
+            HandleBackPress()
             View(modifier)
         }
     }
 
-    /** Derived classes can declare functional (non-ui) Composable blocks before [View()] is invoked. */
     @Composable
-    protected open fun DerivedSetup() {
+    private fun HandleBackPress() {
+        val backPressHandlerPlugins = remember { plugins.filterIsInstance<BackPressHandler>() }
+        val dispatcher = LocalOnBackPressedDispatcherOwner.current?.onBackPressedDispatcher
+        DisposableEffect(dispatcher) {
+            backPressHandlerPlugins.forEach { plugin ->
+                if (!plugin.isCorrect()) {
+                    Appyx.reportException(
+                        IllegalStateException("Plugin $plugin has implementation for both BackPressHandler properties, implement only one")
+                    )
+                }
+                plugin.onBackPressedCallbacks.forEach { dispatcher?.addCallback(it) }
+            }
+            onDispose {
+                backPressHandlerPlugins.forEach { plugin ->
+                    plugin.onBackPressedCallbacks.forEach { it.remove() }
+                }
+            }
+        }
     }
 
+    // BackPressHandler is correct when only one of its properties is implemented.
+    private fun BackPressHandler.isCorrect(): Boolean =
+        onBackPressedCallback != null && listOf(onBackPressedCallback) == onBackPressedCallbacks ||
```

Comment:
`onBackPressedCallback == onBackPressedCallbacks.firstOrNull()`?

Reply:
If `onBackPressedCallbacks` has more than one element, then it means that you override both.

Reply:
Was suggesting avoiding creating an object just for the comparison

Reply:
`onBackPressedCallbacks.size == 1 && onBackPressedCallbacks[0] == onBackPressedCallback`